In [1]:
import pandas as pd 

In [2]:
data=pd.read_csv("thanksgiving.csv",encoding="Latin-1")

In [3]:
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


In [4]:
data.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [5]:
data["Do you celebrate Thanksgiving?"].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [6]:
data=data[data["Do you celebrate Thanksgiving?"]=="Yes"]

In [7]:
data["Do you celebrate Thanksgiving?"].value_counts()

Yes    980
Name: Do you celebrate Thanksgiving?, dtype: int64

In [8]:
data["What is typically the main dish at your Thanksgiving dinner?"].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [9]:
#Display the "Do you typically have gravy?"" column
data[data["What is typically the main dish at your Thanksgiving dinner?"]=="Tofurkey"]["Do you typically have gravy?"]

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [10]:
apple_isnull=data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple"].isnull()

In [11]:
apple_isnull

0       False
1       False
2       False
3        True
4       False
5        True
6       False
7        True
8       False
9        True
11      False
12       True
13      False
14       True
15       True
16       True
17       True
18      False
19      False
20       True
21       True
23      False
24       True
25      False
26      False
27      False
28      False
29      False
30       True
32      False
        ...  
1024     True
1025    False
1026    False
1027     True
1029    False
1030    False
1031    False
1033     True
1034     True
1035     True
1037     True
1038    False
1039    False
1040    False
1041    False
1042    False
1043     True
1044    False
1045     True
1046    False
1047    False
1048     True
1049    False
1050    False
1051     True
1053     True
1054     True
1055    False
1056     True
1057     True
Name: Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple, Length: 980, dtype: bool

In [12]:
pumpkin_isnull=data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin"].isnull()

In [13]:
pecan_isnull=data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"].isnull()

In [14]:
no_pies=apple_isnull&pumpkin_isnull&pecan_isnull

In [15]:
no_pies.value_counts()

False    876
True     104
dtype: int64

In [16]:
def convert_age(row):
    if pd.isnull(row):
        return None
    else:
        row=row.replace("+","") 
        string=row.split(" ")
        age=string[0]
    return(int(age))
data["int_age"]=data["Age"].apply(convert_age)

In [17]:
data.describe()

RespondentID     int_age
count  9.800000e+02  947.000000
mean   4.336737e+09   40.089757
std    4.861132e+05   15.352014
min    4.335895e+09   18.000000
25%    4.336368e+09   30.000000
50%    4.336802e+09   45.000000
75%    4.337009e+09   60.000000
max    4.337955e+09   60.000000

==> we took only the lower bound of each age interval
==> although we have a rough approximation this could be a true depiction of the ages of survey partipants

In [18]:
def convert_income_value(row):
    if pd.isnull(row):
        return None
    elif row.split(" ")[0]=="Prefer":
        return None
    else: 
        row=row.replace("$","")
        row=row.replace(",","")
        string=row.split(" ")
        income_value=string[0]
    return(int(income_value))
data["int_income"]=data["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(convert_income_value)
data.describe()
    

RespondentID     int_age     int_income
count  9.800000e+02  947.000000     829.000000
mean   4.336737e+09   40.089757   75965.018094
std    4.861132e+05   15.352014   59068.636748
min    4.335895e+09   18.000000       0.000000
25%    4.336368e+09   30.000000   25000.000000
50%    4.336802e+09   45.000000   75000.000000
75%    4.337009e+09   60.000000  100000.000000
max    4.337955e+09   60.000000  200000.000000

==> After choosing the lower bound in the interval of income values seems to be very high and  out of the interval [std,mean]  ==> this can't be a true depiction of survey participants 

In [24]:
data[data['int_income']<150000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
Name: How far will you travel for Thanksgiving?, dtype: int64

In [25]:
data[data['int_income']>150000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         49
Thanksgiving is local--it will take place in the town I live in                     25
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    16
Thanksgiving is out of town and far away--I have to drive several hours or fly      12
Name: How far will you travel for Thanksgiving?, dtype: int64

==> Seems like people who earn more then 150000 are more likely to travel then those who learn less 
==> Those who earn less then 150000 spend thanksgiving in the town they live in

In [27]:
data.pivot_table(index="Have you ever tried to meet up with hometown friends on Thanksgiving night?", 
    columns='Have you ever attended a "Friendsgiving?"',
    values="int_age"
)

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [28]:
data.pivot_table(index="Have you ever tried to meet up with hometown friends on Thanksgiving night?", 
    columns='Have you ever attended a "Friendsgiving?"',
    values="int_income"
)

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

==>Poeple with average age 33.976744 (young poeple) tend to spend thanksgiving with hometown friends